bin/i18n/sync-in.rb

```diff
@@ -42,17 +43,66 @@ def localize_docs
   puts "Preparing docs content"
   docs_content_file = File.join(I18N_SOURCE_DIR, "docs", "programming_environments.json")
   programming_env_docs = {}
-  ProgrammingEnvironment.all.each do |env|
-    name = env.name
-    programming_env_docs[name] = {
+  # For each programming environment, name is used as key, title is used as name,
+  ProgrammingEnvironment.all.sort.each do |env| # env as short for environment
+    env_name = env.name
+    programming_env_docs[env_name] = {
       'name' => env.properties["title"],
       'description' => env.properties["description"],
       'categories' => {},
     }
-    env.categories_for_navigation.each do |category|
-      programming_env_docs[name]["categories"].store(
-        category[:key], {'name' => category[:name]}
+    # Programming environment has a method defined in the programming_environment model that returns the
+    # categories for navigation. The method is used to obtain the current categories existing in the database.
+    env.categories_for_navigation.each do |cat| # cat as short for category
+      cat_key = cat[:key]
+      programming_env_docs[env_name]["categories"].store(
+        cat_key, {
+          'name' => cat[:name],
+          'expressions' => {}
+        }
       )
+      cat[:docs].each do |expression|
+        exp_key = expression[:key] # exp_key as short for expression key
+        # Skip javalab
+        next if env_name == 'javalab'
```

Comment:
not sure if this is intentional logic so asking here.

If you skip here, you would have orphan env/categories without expressions (since they are not being added to the dictionary, but the parent keys were added.

Reply:
I am skipping javalab because it doesn't follow the same structure in the database (and javalab is not included in CSD, the current sprint). However, some of the content (the programming categories) can be translated. The idea was to add javalab later to the same file, but it could all be omitted at the beginning.